pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -452,7 +452,7 @@ func (r *resourceManager) ensureDeployment(ctx context.Context) error {
 		return err
 	}
 
-	priorityClassName := v1beta1constants.PriorityClassNameSeedSystemCritical
+	priorityClassName := "gardenlet"
```

Comment:
Why we need this change? Isn't possible GRM to use `gardener-system-critical`?

Reply:
I changed this commit. We can't use `gardener-system-critical` priority class for resource manager now as the deployment will not come up in our e2e test because `gardener-system-critical` is not getting created anywhere. This will cause our e2e test to fail.

Reply:
Once cherry-pick PR is merged. In 1.53 we can use some other priority class instead of `gardener-system-critical` and drop it from managed resource and in 1.54 switch it back to helm deployment.

Reply:
> We can't use gardener-system-critical priority class for resource manager now as the deployment will not come up in our e2e test because gardener-system-critical is not getting created anywhere

In that case, there is istio components using this priority class as well. 

But won't be enough to just change the gardenlet priority class alone: 
1. Because gardenlet once it comes up will deploy `gardener-system-critical`
2. Resource manager can ignore the chart and cleanup the "managed" priorityclass
?

Reply:
> In that case, there are istio components using this priority class as well. 

For that MR will be created and reconciled as GRM will be there.

> Make the gardenlet deploy gardener-system-critical priorityclass once it comes up

We can't make sure when the MR priority class will be cleaned/deleted by GRM meanwhile if gardenlet will try to create one it will throw an error.

Reply:
> We can't make sure when the MR priority class will be cleaned/deleted by GRM meanwhile if gardenlet will try to create one it will throw an error.

Ignore "already exists" error?

Reply:
On a call with @acumino, maybe
"Once cherry-pick PR of this PR is merged. A new PR can be opened for 1.53 release to use some other priority class instead of gardener-system-critical and drop it from managed resource and deploy through helm and in 1.54 switch it back to gardener-system-critical."
is the way to go.

Reply:
Okay, I now got why we need the special handling for GRM - when we bootstrap a new Seed we will fail to create the GRM Deployment because the PriorityClass will be missing (to be deployed by GRM itself later). This reverts the GRM adaptation from https://github.com/gardener/gardener/pull/6257. But for the v1.52 cherry-pick we will need only pure revert of https://github.com/gardener/gardener/pull/6235, right?

I think I am fine with this approach. @timebertt @kris94 comments/objections?

Reply:
No objections from my side, looks like you got the whole scenario.